### PR TITLE
MWPW-142003: Mini Compare Chart Mobile styling

### PR DESCRIPTION
**Pull request by Axelcureno**
Introduces mobile styles for Mini Compare Chart card and an event listener for tab button clicks in order to trigger resize the cards not showing in the DOM.

Screenshot 2024-02-13 at 12 31 12 PM
Screenshot 2024-02-13 at 12 31 02 PM

Tacocat PR-related: https://git.corp.adobe.com/wcms/tacocat.js/pull/538

(resubmitted [PR](https://github.com/adobecom/milo/pull/1887)

Resolves: [MWPW-142003](https://jira.corp.adobe.com/browse/MWPW-142003) and [MWPW-142876](https://jira.corp.adobe.com/browse/MWPW-142876)

**Test URLs:**
- Before: https://main--milo--axelcureno.hlx.page/drafts/axel/block-samples/fragments/mini-compare-chart?martech=off
- After: https://mwpw-142003--milo--axelcureno.hlx.page/drafts/axel/block-samples/fragments/mini-compare-chart?martech=off

### CODE REVIEW

**Comment by Roycethan** [comment]
@Axelcureno Reviewing https://mwpw-142003--milo--axelcureno.hlx.page/drafts/axel/block-samples/fragments/mini-compare-chart few observations noting here:
1) Mobile: CTA text is not center aligned
2)Mobile: Left/Right padding for pricing text is correct ( expected: 16 px) where as for body text its 24 px and CTA its 8 px 
3)Mobile:Theres additional padding below the free-trial CTA
4)Mobile: thre's also an unwanted 16 px padding between Pricing text and promo text
Also tablet styles need to be adjusted if its applicable to this PR, didn't see picker and tabs to check re-size

**Comment by Roycethan** [comment]
> > @Axelcureno Reviewing https://mwpw-142003--milo--axelcureno.hlx.page/drafts/axel/block-samples/fragments/mini-compare-chart few observations noting here:
> > 
> > 1. Mobile: CTA text is not center aligned
> >    2)Mobile: Left/Right padding for pricing text is correct ( expected: 16 px) where as for body text its 24 px and CTA its 8 px
> >    3)Mobile:Theres additional padding below the free-trial CTA
> >    4)Mobile: thre's also an unwanted 16 px padding between Pricing text and promo text
> >    Also tablet styles need to be adjusted if its applicable to this PR, didn't see picker and tabs to check re-size
> 
> @Roycethan please make sure to refresh the page if testing in mobile or resizing the page since this card requires manual resizing by JS code. I also recommend to test from this URL to get a more realistic view https://main--dc--adobecom.hlx.page/dc-shared/fragments/merch-cards/acrobat-business?milolibs=MWPW-142003--milo--axelcureno since milo-forked URLs do not resolve offers (CORS issue) (prices, quantity selectos and offer selectors) causing extra space to show. Also please note that as per the requirement (mentioned in the ticket) each element inside the card should be perfectly aligned horizontally, which can cause extra space to be added in some cards. The padding values (16px or 24px) are assumed if all cards have the same lenght of content.

@Axelcureno Updated my comments based on the above url which i checked for standard mobile resolution say Iphone 12 pro 390*844 px plz review thanks

**Comment by VKniaz** [comment]
Please add tests.

File: libs/blocks/merch-card/merch-card.js [hunk 1]
```diff
@@ -111,13 +111,22 @@ const MULTI_OFFER_CARDS = ['plans', 'product', MINI_COMPARE_CHART];
...
   entries.forEach((entry) => {
-    const container = entry.target.closest('main > div');
+    const container = entry.target.closest('main > div:not([data-status])');
     if (!container) return;
     [...container.querySelectorAll('merch-card')].forEach((card) => card.requestUpdate());
     intersectionObserver.unobserve(entry.target);
   });
 });
 
+const addTabClickListener = (container) => {
```
Comment (yesil) [inline libs/blocks/merch-card/merch-card.js:121]: I don't think adding an event listener is needed.
A merch card that is inside a tab that is hidden is fine,  the intersectionObserver should allow us to detect when the merch-card inside the tab becomes visible when the tab is activated.

Comment (yesil) [reply libs/blocks/merch-card/merch-card.js:121]: I suggest to simplify as follows, feel free to merge proposed change.

https://github.com/Axelcureno/milo/pull/6

test: https://main--cc--adobecom.hlx.page/cc-shared/fragments/merch/products/illustrator/mini-compare/segment-blade?milolibs=mwpw-142003-test--milo--yesil

Comment (Axelcureno) [reply libs/blocks/merch-card/merch-card.js]: Correct, event listener removed. Thank you.

File: test/blocks/merch-card/merch-card-observer.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { readFile } from '@web/test-runner-commands';
+import { expect } from '@esm-bundle/chai';
+
+const sinon = require('sinon');
+
+describe('IntersectionObserver callback', async () => {
```
Comment (yesil) [inline test/blocks/merch-card/merch-card-observer.test.js:6]: I'm not sure whether this test is really needed, and it is currently failing.
If you want to test the intersectionObserver, you can render a hidden card, and make it visible after.
Wait like 50ms and and check that, your css variables are defined.
I would keep this test case in merch-card.test.js though.

**Review by yesil — COMMENTED** [review]
Once the lint and unit test issues are fixed, it is good to merge for me.

**Review by Roycethan — APPROVED** [review]
Left/right padding is fixed/ pricing alignment is fixed, 
Discussed with @Axelcureno regarding additional padding around pricing and CTA's which as per him causing due to making CT's and prices in line with other cards - and the layout is being approved by stakeholders, Per this agreement approving this.

**Pull request merged**